[main] Upgrade to latest dependencies

knative-automation opened the pull request:
Cron -knative-prow-robot

/cc we don't use Prow
/assign we don't use Prow

Produced by: knative-sandbox/knobots/actions/update-deps

Details:
```
bumping k8s.io/kube-openapi e816edb...4241196:
  > 4241196 Merge pull request # 276 from Garima-Negi/fix-owners-file
  > a045323 Merge pull request # 272 from DangerOnTheRanger/openapi-v3-lazy-marshaling
  > 9765b7f Fix typo in OWNERS file
  > c1fc1de Merge pull request # 252 from austince/feat/decouple-restful
  > 8094b93 Add public documentation for HandlerCache.
  > 0d48a34 Merge pull request # 273 from hasheddan/fieldpath-rec
  > 0c8fa98 Testutil cleanup
  > 2dece3d Document that HandlerCache.New is not thread-safe.
  > 258b46d Remove unused code paths in object validator
  > 35b5ddc Add multiple status returns to int tests
  > 4e5b3ca Move HandlerCache to internal package & make HandlerCache tests external.
  > 16e8b81 Set path for schema validators embedded in schema props validators
  > 41ddd1e Take address of local loop var in route status code adapter
  > 9770bc8 Rename cache.go and cache_test.go.
  > 4c65bda Use JSONPath syntax for path indices
  > e9ebba8 Decouple integretion tests from GOPATH
  > 390a58b Remove toGzip from handler3/handler.go.
  > d5a90d6 Set path for all specific validators when set on SchemaValidator
  > eef04e7 Decouple REST Framework from v3
  > cb03b72 Add lazy marshaling for OpenAPIv3 handler.
  > c4d1aeb Decouple REST Framework
```

evankanderson commented:
It looks like the e2e tests keep failing on the `kubeclt wait` -- it seems like we might need more diagnostics to figure out why this is failing?

gabo1208 commented:
It seems so, tried retrying to see if it was a job flake or something

embano1 commented:
I have no idea what I was thinking :)

https://github.com/vmware-tanzu/sources-for-knative/issues/422

Will fix

knative-automation committed:
upgrade to latest dependencies

bumping knative.dev/client 1bab820...736c7c2:
  > 736c7c2 test: use `T.TempDir` to create temporary test directory (# 1660)
  > ff097e9 upgrade to latest dependencies (# 1659)
  > d8ab3d5 upgrade to latest dependencies (# 1658)
  > 9154f26 Fix build warning from go version upgrade (# 1657)
  > b858dab Added subpath functionality to --mount flag (# 1655)
  > 7e8a672 upgrade to latest dependencies (# 1656)
  > bb055ee Added scale-metric flag to configure metric name (# 1653)
  > 76f17f6 Upgrade Go to 1.17 (# 1654)
  > 1cb29fb upgrade to latest dependencies (# 1652)
  > 2929f38 Added flag to configure wait-window between intermediate errors durin… (# 1645)
  > 0ac405d Add timeout option to service create (# 1643)
  > bb7fd73 Added pull-policy flag to service (# 1644)
  > 0d4e83b Update actions (# 1651)
  > b4e95db upgrade to latest dependencies (# 1650)
  > 0993e23 Align codecov config with rest of the org (# 1649)
  > e3d4191 Update actions (# 1648)
  > d07b600 upgrade to latest dependencies (# 1642)
  > b702b42 Fix latest release CI setup (# 1647)
  > 5daa09d Update community files (# 1646)
  > 26ba0e2 Update community files (# 1641)
  > e074fff Update actions (# 1639)
  > f8129fe upgrade to latest dependencies (# 1638)
  > 6170c6d Update community files (# 1640)
  > 73b2d8b Update actions (# 1637)
  > 38b1034 Update community files (# 1636)
bumping gotest.tools/v3 568bc57...dc5149e:
  > dc5149e Merge pull request # 224 from DominicLavery/main
  > 7fe928e Merge pull request # 223 from seriousben/fix-golden-dir-creation
  > 7501e42 switch darwin defaults using runtime.GOOS
  > 10e310b Merge pull request # 222 from gotestyourself/fs-path-existing
  > b0ea9a7 Stop creating directory outside of testdata
  > aca5a3c Fix # 141 comparing symlink permissions
  > 196ee67 Merge pull request # 208 from dnephin/assert-error-is
  > cb4623d fs: add DirFromPath
  > 524f2ce Merge pull request # 221 from dnephin/go1.17
  > bc5434b Remove commented out line
  > 4fbf292 Merge pull request # 218 from thaJeztah/secure_exec
  > 4d8cea6 ci: add go1.17, remove go1.13
  > 567b2a0 lint: ignore line length in tests
  > fd36aec Merge pull request # 219 from dnephin/fix-windows-ci
  > 66ae1b5 icmd: replace all usages of os/exec with golang.org/x/sys/execabs
  > ae5beef assert: Add ErrorIs
  > c280cb4 Merge pull request # 217 from thaJeztah/bump_deps
  > 2dd0fa3 ci: set path for go install
  > cda9c22 go.mod: golang.org/x/tools v0.1.0
  > e419626 poll: small improvement to check functions
  > 3fcbc0e Merge pull request # 216 from thaJeztah/bump_go_16
  > 77bb3d3 go.mod: github.com/google/go-cmp v0.5.5
  > 1df0ee5 Merge pull request # 215 from dnephin/mkdir-in-update
  > d67588d CircleCI: update to GA version of Go 1.16
  > 059fb72 go.mod: github.com/pkg/errors v0.9.1
  > de21663 Merge pull request # 213 from dnephin/ci-go-1.16
  > fbb0800 golden: only create dir if update flag is set
  > 84d691d Merge pull request # 211 from dnephin/add-merge-main-workflow
  > eddfe6c ci: add go1.16
  > 4190a0e Add github action for merging main branch
bumping golang.org/x/oauth2 d3ed0bb...ee48083:
  > ee48083 go.mod: update vulnerable net library
bumping knative.dev/hack f067737...6ffd841:
  > 6ffd841 Update community files (# 168)
  > 02c525c Update community files (# 167)
  > 0e0784b Update community files (# 166)
  > a75ca49 Update community files (# 165)
  > 9c0ea69 Update community files (# 164)
  > c7a1ce1 Update community files (# 163)

Signed-off-by: Knative Automation <automation@knative.team>

evankanderson reviewed (approved):
Looks like automation is now passing, merging.
